src/Bicep.Local.Deploy/Extensibility/LocalExtensibilityHandler.cs

```diff
@@ -37,26 +42,32 @@ public LocalExtensibilityHandler(IModuleDispatcher moduleDispatcher, Func<Uri, T
         RegisteredProviders[new("LocalNested", "0.0.0")] = new AzExtensibilityProvider(this);
     }
 
-    private async Task<ExtensibilityOperationResponse> CallProvider(string method, IExtensibilityProvider provider, ExtensibilityOperationRequest request, CancellationToken cancellationToken)
+    public async Task<ResourceResponseBody> CallExtensibilityHost(
+        string extensionName,
+        string extensionVersion,
+        string method,
+        HttpContent content,
+        CancellationToken cancellationToken)
     {
-        return method switch
-        {
-            "get" => await provider.Get(request, cancellationToken),
-            "delete" => await provider.Delete(request, cancellationToken),
-            "save" => await provider.Save(request, cancellationToken),
-            "previewSave" => await provider.PreviewSave(request, cancellationToken),
-            _ => throw new NotImplementedException($"Unsupported method {method}"),
-        };
+        var provider = RegisteredProviders[new(extensionName, extensionVersion)];
+
+        return await CallProvider(method, provider, content, cancellationToken);
     }
 
-    public async Task<ExtensibilityOperationResponse> CallExtensibilityHost(
+    private async Task<ResourceResponseBody> CallProvider(
         string method,
-        ExtensibilityOperationRequest request,
+        LocalExtensibilityProvider provider,
+        HttpContent content,
         CancellationToken cancellationToken)
     {
-        var provider = RegisteredProviders[new(request.Import.Provider, request.Import.Version)];
-
-        return await CallProvider(method, provider, request, cancellationToken);
+        return method switch
+        {
+            "get" => await provider.Get(await content.ReadAsAsync<ResourceReferenceRequestBody>(cancellationToken), cancellationToken),
+            "delete" => await provider.Delete(await content.ReadAsAsync<ResourceReferenceRequestBody >(cancellationToken), cancellationToken),
+            "createOrUpdate" => await provider.CreateOrUpdate(await content.ReadAsAsync<ResourceRequestBody>(cancellationToken), cancellationToken),
+            "preview" => await provider.Preview(await content.ReadAsAsync<ResourceRequestBody>(cancellationToken), cancellationToken),
```

Comment:
nit: We can elide `async` and `await` since the method is basically a passthrough.

Reply:
Challenge here is that I need to await reading `content` therefore I need to add async / await

Reply:
Ah good point I missed that.